File: Selenium/Activities/Activity10.java
package activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.io.IOException;
import java.time.Duration;

public class Activity10 {
    public static void main(String[] args) throws IOException {
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "/dev/null");
        WebDriverManager.firefoxdriver().setup();
        WebDriver driver = new FirefoxDriver();
        driver.get(" https://v1.training-support.net/selenium/dynamic-controls");
        WebDriverWait wait=new WebDriverWait(driver, Duration.ofSeconds(10));

        System.out.println("Title of webpage:" + driver.getTitle());
        WebElement checkBox=driver.findElement(By.xpath("//input[@class='willDisappear']"));
        if(checkBox.isDisplayed()){
            System.out.println("Checkbox is displayed.");
        }
        else{
            System.out.println("Checkbox is not displayed.");
        }
        driver.findElement(By.id("toggleCheckbox")).click();
        if(checkBox.isDisplayed()){
            System.out.println("Checkbox is displayed.");
        }
        else{
            System.out.println("Checkbox is not displayed.");
        }
        driver.close();
    }
}

File: Selenium/Activities/Activity10.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from selenium.webdriver.support.wait import WebDriverWait
from webdriver_manager.firefox import GeckoDriverManager

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    driver.get("https://v1.training-support.net/selenium/dynamic-controls")
    wait = WebDriverWait(driver, 10)
    print("Title of webpage:" + driver.title)
    checkBox = driver.find_element(By.XPATH, "//input[@class='willDisappear']")
    if checkBox.is_displayed():
        print("Checkbox is displayed.")
    else:
        print("Checkbox is not displayed.")
    driver.find_element(By.ID, "toggleCheckbox").click()
    if checkBox.is_displayed():
        print("Checkbox is displayed.")
    else:
        print("Checkbox is not displayed.")
    driver.close()
